library/std/src/collections/mod.rs

```diff
@@ -401,9 +401,10 @@
 #![stable(feature = "rust1", since = "1.0.0")]
 
 #[stable(feature = "rust1", since = "1.0.0")]
-#[rustc_deprecated(reason = "moved to `std::ops::Bound`", since = "1.52.0")]
+// FIXME(#82080) The deprecation here is only theoretical, and does not actually produce a warning.
+#[rustc_deprecated(reason = "moved to `std::ops::Bound`", since = "1.26.0")]
```

Comment:
I wonder if we could tweak things slightly so that it does work - also see https://github.com/rust-lang/rust/issues/83248#issuecomment-802280311.

The information of which imports you went through *should* exist during name resolution, and we'd only have to keep track of one `DefId` (the "nearest" `pub use` to the import) for deprecation checking purposes.

Reply:
Does that seem like an especially hard change? I only have rudimentary compiler knowledge, but if it's not a monumental task then I'd be willing to tackle it.